src/ds/nonstd.h

```diff
@@ -62,6 +62,14 @@ namespace nonstd
   template <class T>
   using remove_cvref_t = typename remove_cvref<T>::type;
 
+  /** a more generic std::string member function is present in C++20
+   */
+  static inline bool starts_with(
+    const std::string& s, const std::string& prefix)
+  {
+    return s.rfind(prefix, 0) == 0;
```

Comment:
Minor: We have a few other places in the code that do this inline. Can you please replace them with `nonstd::starts_with`?